production/helm/loki/templates/tokengen/job-tokengen.yaml

```diff
@@ -45,6 +45,10 @@ spec:
           image: {{ template "loki.image" . }}
           imagePullPolicy: {{ .Values.loki.image.pullPolicy }}
           args:
+            # The shared emptyDir exists only while the job is running, and deleted once the job is completed.
+            # The tokengen generates a new admin token in case the 'token-file' file doesn't exist.
+            # As a result, subsequent executions of this tokengen job will generate new admin tokens.
+            # Note that previously generated tokens remain valid.
```

Comment:
This means that each installation will generate a new admin token which will be stored in the object storage. The Kubernetes secret will be updated to the latest generated token. 

For instance, if 3 helm installations take place, there will be 3 valid admin tokens stored in the object storage. The Kubernetes secret will be updated with the latest generated token. All these tokens will remain valid.